src/main/java/duke/Lister.java

```diff
@@ -0,0 +1,117 @@
+package duke;
+
+/**
+ * The Lister Class parses commands and references TaskList and Storage Classes to store and retrieve Tasks
+ * @author qiujingying
+ * @version 1.0
+ */
+public class Lister {
+    private TaskList taskList;
+    private Storage storage;
+
+    /**
+     * Creates a Lister object to reference to TaskList and Storage objects
+     * @param tasks initial TaskList stores Tasks
+     * @param storage initial storage with filepath to store data
+     */
+    public Lister(TaskList tasks, Storage storage) {
+        this.taskList = tasks;
+        this.storage = storage;
+    }
+
+    /**
+     * Parses commands from command line
+     * @param command input to identify relevant Task objects to create and their description
+     */
+    public void record(String command) {
+        if (command.equals("list")) {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < taskList.getSize(); i++) {
+                System.out.println((i + 1) + "." + taskList.retrieveTask(i).toString());
+            }
+        } else {
+            try {
+                int x = getIndex(command);
+                switch (command.substring(0, x)) {
+                case "delete":
+                    int b = Integer.valueOf(command.substring(x + 1)) - 1;
+                    taskList.removeTask(b);
+                    System.out.println("Noted. I've removed this task:\n" + taskList.retrieveTask(b).toString());
+                    System.out.println("Now you have " + taskList.getSize() + " tasks in the list.");
+                    break;
+                case "done":
+                    int y = Integer.valueOf(command.substring(x + 1)) - 1;
+                    taskList.retrieveTask(y).markAsDone();
+                    System.out.println("Nice! I've marked this task as done:\n" + taskList.retrieveTask(y).toString());
+                    break;
+                case "deadline":
+                    int z = command.indexOf('/');
```

Comment:
What if the user inputs a task description that contains a forward slash '/'? Will your parser be able to parse it?

For example, what will happen if the user inputs: deadline read **2/3** chapters of textbook **/by** <date>